src/main/java/com/okta/developer/security/oauth2/ScimGroupService.java

```diff
@@ -87,6 +96,40 @@ public FilterResponse<ScimGroup> find(Filter filter, PageRequest pageRequest, So
 
     @Override
     public void delete(String s) throws ResourceException {
-        log.debug("delete...");
+        authorityRepository.delete(new Authority(s));
+    }
+
+    private ScimGroup createOrUpdateGroup(ScimGroup scimGroup) {
+        String groupId = scimGroup.getDisplayName();
+        scimGroup.setId(groupId);
+        authorityRepository.save(new Authority(groupId));
+
+        scimGroup
+            .getMembers()
+            .stream()
+            .forEach(memberRef -> {
+                // Assume these are always userIds, but per spec they could be groupIds (Okta only supports users here)
+                String id = memberRef.getValue();
+                Optional<User> optionalUser = userRepository.findById(id);
+                optionalUser.ifPresentOrElse(
+                    user -> {
+                        user.getAuthorities().add(new Authority(groupId));
+                        // TODO: user cache needs to be invalidated, so new groups/authorities are loaded
```

Comment:
I'm guessing we could just remove the Cache annotation from the UserRepo class.

I'm not sure how far we want to go down this road for the demo.
(either way, I want to keep hacking on this afterward)